Homeworks/Homework2/src/main/demoblaze_page.py

```diff
@@ -0,0 +1,67 @@
+import time
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+class DemoBlazePage:
+    user_name = (By.ID, "loginusername")
+    user_password = (By.ID, "loginpassword")
+    login_button = (By.ID, "login2")
+    login_text_button = (By.XPATH, "//button[contains(text(),'Log in')]")
+    logout_button = (By.ID, "logout2")
+    welcome_message = (By.ID, "nameofuser")
+
+    monitors_button = (By.LINK_TEXT, "Monitors")
+
+    def __init__(self, driver):
+        self.driver = driver
+        self.wait = WebDriverWait(self.driver, 3)
+
+    def chrome_driver(self):
+        self.driver = webdriver.Chrome()
+
+    def open_url(self):
+        self.driver.get("https://www.demoblaze.com/")
+
+    def click_login_button(self):
+        self.driver.find_element(*self.login_button).click()
+
+    def setup_login_and_password(self, userlogin, password):
+        self.wait.until(EC.visibility_of_element_located(self.user_name)).send_keys(userlogin)
+
+        self.driver.find_element(*self.user_password).send_keys(password)
+        self.driver.find_element(*self.login_text_button).click()
+        time.sleep(3)
+
+    def verify_login(self, login):
+        self.wait.until(EC.visibility_of_element_located(self.welcome_message))
+        welcome_message = self.driver.find_element(*self.welcome_message).text
+        assert welcome_message == f"Welcome {login}"
+
+    def click_monitors_button(self):
+        self.driver.find_element(*self.monitors_button).click()
+
+    def select_high_price_item(self):
+        monitor_items = self.driver.find_elements(By.CLASS_NAME, "card-block")
+        time.sleep(3)
```

Comment:
it would be better avoid using sleep and wait for a specific event to check it's state instead

Reply:
thanks, changed